contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1180,6 +1191,133 @@ tle_name_comparison(const char *tlename, const char *identifier)
 		return (0 == strcmp(tlename, identifier));
 }
 
+/*
+* A special case of the get_object_address_relobject() function, specifically
+* for the case of triggers in tsql dialect. We add a pg_trigger lookup to search
+* for the relation that the trigger is associated with, since the relation name
+* is not supplied by the user, and thus not a part of the *object list.
+*/
+static ObjectAddress
+get_trigger_object_address(List *object, Relation *relp, bool missing_ok)
+{
+	ObjectAddress address;
+	Relation	relation = NULL;
+	const char *depname;
+	Oid			reloid;
+	Relation	tgrel;
+	ScanKeyData		key;
+	SysScanDesc tgscan;
+	HeapTuple	tuple;
+	const char	*trigger_physical_schema = NULL;
+	const char	*trigger_logical_schema = NULL;
+	const char	*cur_dbo_physical_schema= NULL;
+	bool 	found_trigger = false;
+	const char	*pg_trigger_physical_schema = NULL;
+	const char	*pg_trigger_logical_schema = NULL;
+	const char	*cur_physical_schema = NULL;
+
+	if (sql_dialect != SQL_DIALECT_TSQL)
+	{
+		address.classId = InvalidOid;
+		address.objectId = InvalidOid;
+		address.objectSubId = InvalidAttrNumber;
+		return address;
+	}
+	/* Extract name of dependent object. */
+	depname = strVal(llast(object));
+	if (list_length(object) > 1){
+		trigger_physical_schema = ((Value *)list_nth(object,0))->val.str;
+		trigger_logical_schema = get_logical_schema_name(trigger_physical_schema, true);
+	}
+
+	cur_dbo_physical_schema = get_dbo_schema_name(get_cur_db_name());
+
+	if (prev_get_trigger_object_address_hook)
+		return (*prev_get_trigger_object_address_hook)(object,relp,missing_ok);
```

Comment:
Why are we calling previous hook here instead of at the beginning of the function?

Reply:
Move it to the beginning of the function